### DISCUSSION TITLE
Would it be possible to generate an image for an headless Raspberry Pi 0w to use it as an ShredOS machine?

### DISCUSSION THREAD

**Discussion by LeatherMolly** [PartialVolume/shredos.x86_64#101] [Q&A]
I found no possibility to contact the ShredOS dev so I try to write down my idea/ question and hope that the users will treat me nicely ;-)

I have an Raspberry Pi 0w with an usb HAT. My Idea would be to have an ShredOS which can simultanly erease up to 3 HDDs or USB Sticks.

Would this be possible? Even if it would be for only 1 delete process, this would be great. 

Another option could be, that if an hdd or USB Stick is connected the ShredOS would start automatically to perform the delete option chosen before in a config file.

So if it will be possible I would love to get instructions how to build an raspberry Pi 0w image by myself. And it would be nice to have it written for basic users so it could be used by non-IT human beeings ;-) trying to learn more about computing.

Have all a nice day & all the best 2 ya all

P. S. on the other hand it would be nice to get arm/ raspberry pi images to download on this site.

**Comment by PartialVolume** [comment]
>I have an Raspberry Pi 0w with an usb HAT. My Idea would be to have an ShredOS which can simultanly erease up to 3 HDDs or USB Sticks.

I wouldn't recommend it on a Raspberry Pi Zero for performance reasons especially if you are trying to wipe modern drives. While a raspberry PI specific build of ShredOS might be able to run on a RPI zero it's performance especially for random passes would be quite slow as random number generation is quite CPU intensive. As far as I know RPI zero uses USB 2.0 (480 Mbps) which means hard disks that have a max sustained speed of < 60MBps would possibly be ok but anything faster would be restricted to this speed.

>Another option could be, that if an hdd or USB Stick is connected the ShredOS would start automatically to perform the delete option chosen before in a config file.

How to do this is described [here](https://github.com/PartialVolume/shredos.x86_64#how-to-change-the-default-nwipe-options-so-the-change-persists-between-reboots) in the README.md file

>So if it will be possible I would love to get instructions how to build an raspberry Pi 0w image by myself. And it would be nice to have it written for basic users so it could be used by non-IT human beeings ;-) trying to learn more about computing.

I'll leave that to somebody else, as unfortunately I find it hard enough to find time to work on the code for nwipe, so somebody else would need write such documentation.

If I was to put ShredOS on a Raspberry PI, I would probably use a RPI 4 due to it having USB 3.0 ports and because of it's processing power.

>P. S. on the other hand it would be nice to get arm/ raspberry pi images to download on this site.

No plans for that at the moment, although if somebody was to build ShredOS for RPI and test it I would certainly check it out and include a image and documentation. I do have RPI4's for testing but need to focus on the nwipe features that need to be added before working on other builds.

**Reply by LeatherMolly** [reply]
Thanks
